File: Algorithms/python/404 - Sum of Left Leaves.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def sumOfLeftLeaves(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        if not root:
            return 0
            
        return self.visitLeaves(root, False)
        
    def visitLeaves(self, node, is_left):
        # var
        left = 0
        right = 0
        
        if not node.left and not node.right:
            if is_left:
                return node.val
            else:
                return 0
        
        if node.left:
            left = self.visitLeaves(node.left, True) 
        if node.right:
            right = self.visitLeaves(node.right, False)
            
        return left + right
        

File: Algorithms/cpp/404 - Sum of Left Leaves.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    int sumOfLeftLeaves(TreeNode* root) {
        if (!root)
            return 0;
            
        return helper(root, false);
    }
    
    int helper(TreeNode* root, bool is_left) {
        if (!root->left && !root->right)
            return is_left ? root->val : 0;
    
        // var
        int left = !root->left ? 0 : helper(root->left, true);
        int right = !root->right ? 0 : helper(root->right, false);
    
        return left + right;
    }
};